sycl/include/CL/sycl/builtins.hpp

```diff
@@ -1306,6 +1340,8 @@ template <typename T, typename T2>
 detail::enable_if_t<
     detail::is_genfloatf<T>::value && detail::is_genint<T2>::value, T>
 select(T a, T b, T2 c) __NOEXC {
+  static_assert(detail::vector_size<T>::value == detail::vector_size<T2>::value,
+                "");
```

Comment:
What about some message here? I think this would be useful from user experience point of view.

Reply:
I think current error message is quite informative.
```
private_sycl/llvm/build/lib/clang/10.0.0/include/CL/sycl/builtins.hpp:1343:3: error: static_assert failed due to requirement 'detail::vector_size<cl::sycl::vec<float, 3> >::value == detail::vector_size<int>::value' ""
  static_assert(detail::vector_size<T>::value == detail::vector_size<T2>::value,
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
private_sycl/llvm/sycl/test/basic_tests/buffer/buffer_dev_to_dev.cpp:26:22: note: in instantiation of function template specialization 'cl::sycl::select<cl::sycl::vec<float, 3>, int>' requested here
  auto R = cl::sycl::select(A, B, M);
```

Reply:
I can't agree. It's not clear how `T` and `T2` are mapped to the original source code from this message.

Reply:
I will try to write a general error message.
Do you like this option? `The arguments must have/point to the same number of elements and bits`

Reply:
Or for example:
to map with source code/specification i should rename `T` as `geninteger` and `T2` as `igeninteger` for select build-in geninteger select (geninteger a, geninteger b, igeninteger c). And do this for all built-in functions.

Reply:
How about: "agument sizes must be the same"?

Reply:
It is not clear what is the augment size. Is it size of bits or vector element count or both.

For example for `genfloat modf (genfloat x, genfloatptr iptr)` this error message does not work. 
There is a genfloatptr type which must be a mupti_ptr to the same type like genfloat.